version: '3'
services:
  app:
    build: 
      context: node
    image: eduardoward/node
    container_name: app
    tty: true
    networks:
      - node-network
    healthcheck:
      test: curl --fail http://localhost:3000/ || exit 1
      interval: 5s
      timeout: 10s
      retries: 10
    depends_on:
      db:
        condition: service_healthy

  nginx:
    build: 
      context: nginx
      dockerfile: Dockerfile.prod
    image: eduardoward/nginx:prod
    container_name: nginx
    networks:
      - node-network
    ports:
      - "8080:80"
    depends_on:
      app:
        condition: service_healthy

  db:
    build:
      context: db
    image: eduardoward/db
    command: --innodb-use-native-aio=0
    container_name: db
    restart: always
    volumes:
      - ./db/mysql:/var/lib/mysql
      - ./db/init.sql:/docker-entrypoint-initdb.d/init.sql
    environment:
      - MYSQL_DATABASE=nodedb
      - MYSQL_ROOT_PASSWORD=root
    tty: true
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost", "-u", "root", "-p$$MYSQL_ROOT_PASSWORD"]
      interval: 10s
      timeout: 10s
      retries: 10
    networks:
    - node-network

networks:
  node-network:
    driver: bridge